src/model/entities/FormScreenLifecycle/FormScreenLifecycle.tsx

```
@@ -484,6 +489,53 @@ export class FormScreenLifecycle02 implements IFormScreenLifecycle02 {
     yield* this.startAutoRefreshIfNeeded();
   }
 
+  private initializePlugins(initUIResult: any) {
+    let formLevelPlugins = find(initUIResult.formDefinition, (node: any) => node.attributes?.Type === "FormLevelPlugin");
+    if(formLevelPlugins.length > 0){
+      const formScreen = getFormScreen(this);
+      for (let dataView of formScreen.dataViews) {
+        dataView.clear()
+      }
+    }
+    let sessionId = getSessionId(this);
+    formLevelPlugins
+      .forEach(node => {
+        const plugin = pluginLibrary.get(
+          {
+            name: node.attributes.Name,
+            modelInstanceId: node.attributes.ModelInstanceId,
+            sessionId: sessionId
+          });
+        if (!isIFormPlugin(plugin)) {
+          throw new Error(`Plugin ${node.attributes.Name} is not FormLevelPlugin`)
```

Maybe I would let it fail sooner, maybe even during compilation time, maybe even not let this happen in the Architect, because this signals the plugin is installed where it should not... what do you think?

---

This should check that the type of plugin defined in the Architect matches the type in javascript. We need the xml node and the plugin instance to check that so I don't think we can do that any sooner.